fbw-a32nx/src/wasm/systems/a320_systems/src/hydraulic/mod.rs

```diff
@@ -5633,6 +5677,304 @@ impl SimulationElement for A320TrimInputController {
+
+        // TODO: this should come from a SEC output, this is a placeholder of the SEC output
+        self.tertiary_lock_from_sec_should_unlock
+            .update(context, self.throttle_lever_angle.get::<degree>() <= -3.);
+
+        self.state = match self.state {
+            ReverserControlState::StowedOff => {
+                if deploy_authorized
+                    && is_confirmed_stowed_available_for_deploy
+                    && allow_power_to_valves
+                {
+                    ReverserControlState::StowedOn
+                } else {
+                    self.state
+                }
+            }
+            ReverserControlState::StowedOn => {
+                if !allow_power_to_valves {
+                    ReverserControlState::StowedOff
+                } else if allow_isolation_valve_to_open {
+                    ReverserControlState::TransitOpening
+                } else {
+                    self.state
+                }
+            }
+            ReverserControlState::TransitOpening => {
+                if reverser_feedback.proximity_sensor_all_deployed() {
+                    ReverserControlState::FullyOpened
+                } else if !deploy_authorized || !allow_power_to_valves {
+                    ReverserControlState::TransitClosing
+                } else {
+                    self.state
+                }
+            }
+            ReverserControlState::TransitClosing => {
+                if reverser_feedback.proximity_sensor_all_stowed() {
+                    ReverserControlState::StowedOff
+                } else if allow_directional_valve_to_deploy && deploy_authorized {
+                    ReverserControlState::TransitOpening
+                } else {
+                    self.state
+                }
+            }
+            ReverserControlState::FullyOpened => {
+                if !deploy_authorized || !allow_power_to_valves {
+                    ReverserControlState::TransitClosing
+                } else {
+                    self.state
+                }
+            }
+        };
+    }
+}
+impl SimulationElement for A320ReverserController {
+    fn read(&mut self, reader: &mut SimulatorReader) {
+        self.throttle_lever_angle = reader.read(&self.throttle_lever_angle_id);
+    }
+}
+impl ReverserInterface for A320ReverserController {
+    fn should_unlock(&self) -> bool {
+        self.tertiary_lock_from_sec_should_unlock.output()
+    }
+
+    fn should_power_valves(&self) -> bool {
+        self.state != ReverserControlState::StowedOff
+    }
+
+    fn should_isolate_hydraulics(&self) -> bool {
+        self.state == ReverserControlState::StowedOff
+            || self.state == ReverserControlState::StowedOn
+    }
+
+    fn should_deploy_reverser(&self) -> bool {
+        self.state == ReverserControlState::TransitOpening
+            || self.state == ReverserControlState::FullyOpened
+    }
+}
+impl Debug for A320ReverserController {
```

Comment:
We can remove this now.

Reply:
Yeah can be useful sometimes as it's a quick way to print debug console info. And you can nest those debug impl from top to nested objects to quickly get informations you need